src/lib/transition_frontier/transition_frontier.ml

```diff
@@ -209,7 +232,8 @@ module Make (Inputs : Inputs_intf) :
     ; consensus_local_state: Consensus.Local_state.t
     ; extensions: Extensions.t
     ; extension_readers: Extensions.readers
-    ; extension_writers: Extensions.writers }
+    ; extension_writers: Extensions.writers
+    ; root_history: Root_history.t }
```

Comment:
I wonder, should the root history not be an extension itself? It's nice to lift things up into extensions as the core of the transition frontier is already so complex. Up to you though. What do you think?